src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -611,7 +611,7 @@ internal static string ToDebugString(this Expression expr)
 
     internal class FunctionContext
     {
-        internal ScriptBlock _scriptBlock;
+        internal ScriptBlock ScriptBlock;
```

Comment:
per https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md it seems the current style is correct?  Also, I believe only public fields have Pascal Casing otherwise it would be camelCasing

Reply:
The properties is used in other classes and files.
It seems it is bad if we see a code like follow in another file:
```c#
  functionContext._scriptBlock.Ast.Extent
```
 I believe the properties should be _public_ in the _internal_ class. Thoughts?

Reply:
@SteveL-MSFT Could you please continue?

Reply:
I removed all extra changes and refactor commits.